src/main/java/at/ac/tuwien/kr/alpha/common/atoms/AggregateAtom.java

```diff
@@ -0,0 +1,225 @@
+package at.ac.tuwien.kr.alpha.common.atoms;
+
+import at.ac.tuwien.kr.alpha.common.ComparisonOperator;
+import at.ac.tuwien.kr.alpha.common.Predicate;
+import at.ac.tuwien.kr.alpha.common.terms.Term;
+import at.ac.tuwien.kr.alpha.common.terms.VariableTerm;
+import at.ac.tuwien.kr.alpha.grounder.Substitution;
+
+import java.util.*;
+
+import static at.ac.tuwien.kr.alpha.Util.join;
+import static at.ac.tuwien.kr.alpha.Util.oops;
+
+/**
+ * Copyright (c) 2017, the Alpha Team.
+ */
+public class AggregateAtom implements Atom {
+
+	private final ComparisonOperator lowerBoundOperator;
+	private final Term lowerBoundTerm;
+	private final ComparisonOperator upperBoundOperator;
+	private final Term upperBoundTerm;
+	private final AGGREGATEFUNCTION aggregatefunction;
+	private final List<AggregateElement> aggregateElements;
+
+	public AggregateAtom(ComparisonOperator lowerBoundOperator, Term lowerBoundTerm, ComparisonOperator upperBoundOperator, Term upperBoundTerm, AGGREGATEFUNCTION aggregatefunction, List<AggregateElement> aggregateElements) {
+		this.lowerBoundOperator = lowerBoundOperator;
+		this.lowerBoundTerm = lowerBoundTerm;
+		this.upperBoundOperator = upperBoundOperator;
+		this.upperBoundTerm = upperBoundTerm;
+		this.aggregatefunction = aggregatefunction;
+		this.aggregateElements = aggregateElements;
+		if (upperBoundOperator != null || lowerBoundOperator != ComparisonOperator.LE) {
+			throw new UnsupportedOperationException("Aggregate construct not yet supported.");
+		}
+		// TODO: add defaults if some bound is not given!
+	}
+
+	@Override
+	public boolean isGround() {
+		for (AggregateElement aggregateElement : aggregateElements) {
+			for (Term elementTerm : aggregateElement.elementTerms) {
+				if (!elementTerm.isGround()) {
+					return false;
+				}
+			}
+			for (Literal elementLiteral : aggregateElement.elementLiterals) {
+				if (!elementLiteral.isGround()) {
+					return false;
+				}
+			}
+		}
+		if (lowerBoundTerm != null && !lowerBoundTerm.isGround()
+			|| upperBoundTerm != null && !upperBoundTerm.isGround()) {
+			return false;
+		}
+		return true;
+	}
+
+
+	@Override
+	public AggregateLiteral toLiteral(boolean positive) {
+		return new AggregateLiteral(this, positive);
+	}
+
+	@Override
+	public List<Term> getTerms() {
+		throw oops("Aggregate atom cannot report terms.");
+	}
+
+	@Override
+	public Predicate getPredicate() {
+		throw oops("Aggregate atom cannot report predicate.");
+	}
+
+	/**
+	 * Returns all variables occurring inside the aggregate, between { ... }.
+	 * @return each variable occurring in some aggregate element.
+	 */
+	public List<VariableTerm> getAggregateVariables() {
```

Comment:
I think the code would become cleaner if `AggregateElement` would implement `getOccurringVariables()`.

Reply:
Done.